sentry/src/main/java/io/sentry/Hub.java

```diff
@@ -630,4 +632,79 @@ public void flush(long timeoutMillis) {
     }
     return clone;
   }
+
+  @ApiStatus.Internal
+  @Override
+  public SentryId captureTransaction(
+      final @NotNull SentryTransaction transaction, final @Nullable Object hint) {
+    Objects.requireNonNull(transaction, "transaction is required");
+
+    SentryId sentryId = SentryId.EMPTY_ID;
+    if (!isEnabled()) {
+      options
+          .getLogger()
+          .log(
+              SentryLevel.WARNING,
+              "Instance is disabled and this 'captureTransaction' call is a no-op.");
+    } else if (!Boolean.TRUE.equals(transaction.isSampled())) {
+      options
+          .getLogger()
+          .log(
+              SentryLevel.DEBUG,
+              "Transaction %s was dropped due to sampling decision.",
+              transaction.getEventId());
+    } else {
+      try {
+        final StackItem item = stack.peek();
+        if (item != null) {
+          sentryId = item.client.captureTransaction(transaction, item.scope, hint);
+          item.scope.clearTransaction();
```

Comment:
does it make sense to have this on a `finally` block or `captureTransaction` would never throw?

Reply:
I guess it may throw but would we want to clear transaction then or leave it up to users if they prefer to repeat?

Reply:
good question, if it's up to me, I'd clear it anyway, otherwise, every call to `captureTransaction` should be wrapped by the client to capture an exception, it's not a good experience IMO, also the capture methods should never throw iirc, unless we add `throws` to the signature, so they are forced to do it.